pkg/kapp/preflight/registry_test.go

```diff
@@ -106,3 +117,174 @@ func TestRegistryRun(t *testing.T) {
 		})
 	}
 }
+
+func TestRegistryDirectConfig(t *testing.T) {
+	testCases := []struct {
+		name      string
+		registry  *Registry
+		config    map[string]any
+		shouldErr bool
+	}{
+		{
+			name: "preflight checks registered, no config",
+			registry: &Registry{
+				known: map[string]Check{
+					"disabledCheck": NewCheck(func(_ context.Context, _ *diffgraph.ChangeGraph, cfg CheckConfig) error {
+						if cfg != nil {
+							return errors.New("config should not be present")
+						}
+						return nil
+					}, true),
+				},
+			},
+		},
+		{
+			name: "preflight checks registered, config set, no error",
+			registry: &Registry{
+				known: map[string]Check{
+					"someCheck": NewCheck(func(_ context.Context, _ *diffgraph.ChangeGraph, cfg CheckConfig) error {
+						if cfg == nil {
+							return errors.New("config should be present")
+						}
+						v, ok := cfg["foo"]
+						if !ok {
+							return errors.New("foo config not present")
+						}
+						if v != "bar" {
+							return errors.New("foo should equal 'bar'")
+						}
+						return nil
+					}, true),
+				},
+			},
+			config: map[string]any{
+				"foo": "bar",
+			},
+		},
+		{
+			name: "preflight checks registered, unexpected config set, no error",
+			registry: &Registry{
+				known: map[string]Check{
+					"someCheck": NewCheck(func(_ context.Context, _ *diffgraph.ChangeGraph, cfg CheckConfig) error {
+						if cfg == nil {
+							return errors.New("config should be present")
+						}
+						_, ok := cfg["foobar"]
+						if ok {
+							return errors.New("foo config should not present")
+						}
+						return nil
+					}, true),
+				},
+			},
+			config: map[string]any{
+				"foo": "bar",
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			for _, check := range tc.registry.known {
+				_ = check.SetConfig(tc.config)
+			}
+			err := tc.registry.Run(nil, nil)
+			if tc.shouldErr {
+				require.Error(t, err)
+			} else {
+				require.NoError(t, err)
+			}
```

Comment:
I don't think we should be running the checks when unit testing the `Registry.SetConfig` method. I'm imagining something more along the lines of:
```go
for _, tc := range testCases {
    t.Run(tc.name, func(t *testing.T) {
        err := tc.registry.SetConfig(tc.config)
        require.Equal(t, tc.shouldErr, err != nil)
    }
}
```

Reply:
This might require the creation of a more detailed mock of the `preflight.Check` interface that allows us to test in this format, but I think is worthwhile because it lets us test just the `SetConfig` method and allows us to test each "unit" in that logic

Reply:
The mock `preflight.Check` interface is sufficient for testing, since it's `Run()` takes in the configuration, rather than having its `preflight.SetSetConfig()` save the configuration (which is what we'd normally expect it to do).

This would certainly complicate the testing logic to have a "SetConfig()" callback for `preflight.Check()`, but it's doable.

Reply:
Note that this:
```
        require.Equal(t, tc.shouldErr, err != nil)
```
Actually hides what `err` is, and subsequently, makes it harder to debug.